### PR TITLE
rpc: fix test data races

### PR DESCRIPTION
**Pull request by erikgrinaker**
Fixes a couple of test data races. `testing.T` is not concurrency-safe and should not be used in goroutines at all.

### CODE REVIEW

**Review by marbar3778 — APPROVED** [review]
Thank you. 

LGTM

File: rpc/jsonrpc/server/http_server_test.go [hunk 1]
```diff
@@ -57,13 +56,10 @@ func TestMaxOpenConnections(t *testing.T) {
 			c := http.Client{Timeout: 3 * time.Second}
 			r, err := c.Get("http://" + l.Addr().String())
 			if err != nil {
-				t.Log(err)
```
Comment (melekes) [inline rpc/jsonrpc/server/http_server_test.go:60]: The other reporting methods, such as the variations of Log and Error, may be called simultaneously from multiple goroutines.

File: rpc/jsonrpc/server/http_server_test.go [hunk 2]
```diff
@@ -85,12 +81,18 @@ func TestServeTLS(t *testing.T) {
 		fmt.Fprint(w, "some body")
 	})
 
+	chErr := make(chan error, 1)
 	go func() {
-		if err := ServeTLS(ln, mux, "test.crt", "test.key", log.TestingLogger(), DefaultConfig()); err != nil {
-			t.Log(err)
```
Comment (melekes) [inline rpc/jsonrpc/server/http_server_test.go:90]: t.Log is safe

Comment (erikgrinaker) [reply rpc/jsonrpc/server/http_server_test.go:90]: Interesting, the race detector was throwing errors because of this line.

Comment (erikgrinaker) [reply rpc/jsonrpc/server/http_server_test.go:90]: I reran the job so I'm unfortunately not able to get the build output, but will keep this in mind for the next time something similar happens. I may just have misread the goroutine dump or something.

**Pull request merged**